activerecord/lib/active_record/result.rb

```diff
@@ -127,14 +129,22 @@ def cast_values(type_overrides = {}) # :nodoc:
     def initialize_copy(other)
       @columns      = columns.dup
       @rows         = rows.dup
-      @column_types = column_types.dup
       @hash_rows    = nil
+      @types        = types.dup
+    end
+
+    def column_types
+      if @types.present?
```

Comment:
Since `@types` is an array, you can use `any?` instead of `present?` here – or swap the branches and use `empty?`.

Reply:
👍 I made this change.